modules/strelka/1.1/strelka.smk

```diff
@@ -184,46 +184,86 @@ rule _strelka_run:
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_filter_{var_type}.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_filter_{var_type}.stderr.log"
     shell:
         op.as_one_line("""
-        awk 'BEGIN {{FS=OFS="\t"}} $0 ~ /^#/ || $7 == "PASS"' {input.vcf} > {output.vcf}
+        bcftools view -f ".,PASS" -Oz -o {output.vcf} {input.vcf_gz} > {log.stdout} 2> {log.stderr} && tabix -p vcf {output.vcf} >> {log.stdout} 2>> {log.stderr} 
         """)
 
+#merge indel and snv vcf into one combined file for paired jobs
+rule _combine_strelka:
+    input:
+        snv = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/somatic.snvs.passed.vcf.gz",
+        indel = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/somatic.indels.passed.vcf.gz"
+    output:
+        combined = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/combined.passed.vcf.gz",
+        combined_tbi = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/combined.passed.vcf.gz.tbi"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    resources: 
+        mem_mb = CFG["mem_mb"]["bcftools_sort"]
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_combine.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_combine.stderr.log"
+    shell:
+        op.as_one_line("""
+        bcftools concat -a {input.snv} {input.indel} | bcftools sort --max-mem {resources.mem_mb}M -Oz -o {output.combined} > {log.stdout} 2> {log.stderr} && tabix -p vcf {output.combined} >> {log.stdout} 2>> {log.stderr} 
+        """)
 
 # Symlinks the final output files into the module results directory (under '99-outputs/')
-rule _strelka_output_vcf:
+rule _strelka_output_filtered_vcf:
     input:
-        vcf = str(rules._strelka_decompress_vcf.output.vcf),
-        vcf_p = str(rules._strelka_filter.output.vcf)
+        vcf = str(rules._combine_strelka.output.combined),
+        vcf_tbi = str(rules._combine_strelka.output.combined_tbi)
     output:
-        vcf = CFG["dirs"]["outputs"] + "vcf/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{var_type}.vcf",
-        vcf_p = CFG["dirs"]["outputs"] + "vcf/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{var_type}.passed.vcf"
+        vcf = CFG["dirs"]["outputs"] + "vcf/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}_combined.passed.vcf.gz",
```

Comment:
I thought that my use of var_type to represent either "somatic.snvs" or "somatic.indels" (for paired) or "variants" (for unpaired) in the function above would allow them to be linked but I honestly don't have a lot of confidence in that.